exist-core/src/main/java/org/exist/dom/persistent/BinaryDocument.java

```diff
@@ -45,39 +46,45 @@
     private BlobId blobId;
     private long realSize = 0L;
 
-    public BinaryDocument(final BrokerPool pool) {
-        super(pool);
-    }
-
     /**
      * Creates a new persistent binary Document instance.
      *
      * @param pool The broker pool
      * @param collection The Collection which holds this document
+     * @param docId the id of the document
      * @param fileURI The name of the document
      */
-    public BinaryDocument(final BrokerPool pool, final Collection collection, final XmldbURI fileURI) {
-        super(pool, collection, fileURI);
+    public BinaryDocument(final BrokerPool pool, final Collection collection, final int docId, final XmldbURI fileURI) {
```

Comment:
I think for some of my extensions this is an API change.
Not a problem for me as I'd check and refactor for v5.x anyway